lib/jsdom/browser/Window.js

```diff
@@ -401,6 +414,27 @@ function Window(options) {
     writable: true
   });
 
+  // https://dom.spec.whatwg.org/#dom-range-range
+  function Range() {
```

Comment:
Why implement this in this manner instead of using webidl2js?

Reply:
This is not needed anymore, it was used to associate the `Range` with the `Document`. This is not needed anymore with the webidl2js constructor reform. The `Document` can now be extracted out of the newly introduced `globalObject`.